File: src/main/java/com/multicampus/gamesungcoding/a11ymarketserver/feature/user/entity/UserA11yProfile.java
package com.multicampus.gamesungcoding.a11ymarketserver.feature.user.entity;

import com.multicampus.gamesungcoding.a11ymarketserver.common.id.UuidV7;
import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "USER_A11Y_PROFILES")
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@EntityListeners(AuditingEntityListener.class)
public class UserA11yProfile {

    @Id
    @UuidV7
    @Column(nullable = false, updatable = false, length = 16)
    private UUID profileId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false, updatable = false)
    @OnDelete(action = OnDeleteAction.CASCADE)
    private Users user;

    @Embedded
    A11yProfileInfo profileInfo;

    @CreatedDate
    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @LastModifiedDate
    @Column(nullable = false)
    private LocalDateTime updatedAt;

    @Builder
    private UserA11yProfile(
            Users user,
            A11yProfileInfo profileInfo) {

        this.user = user;
        this.profileInfo = profileInfo;
    }

    public void update(A11yProfileInfo profileInfo) {
        this.profileInfo = profileInfo;
    }

}

File: src/main/kotlin/com/multicampus/gamesungcoding/a11ymarketserver/feature/user/entity/UserA11yProfile.kt
package com.multicampus.gamesungcoding.a11ymarketserver.feature.user.entity

import com.multicampus.gamesungcoding.a11ymarketserver.common.id.UuidV7
import jakarta.persistence.*
import org.hibernate.annotations.OnDelete
import org.hibernate.annotations.OnDeleteAction
import org.springframework.data.annotation.CreatedDate
import org.springframework.data.annotation.LastModifiedDate
import org.springframework.data.jpa.domain.support.AuditingEntityListener
import java.time.LocalDateTime
import java.util.*

@Entity
@Table(name = "USER_A11Y_PROFILES")
@EntityListeners(AuditingEntityListener::class)
class UserA11yProfile(
    @Embedded
    var profileInfo: A11yProfileInfo,

    @OnDelete(action = OnDeleteAction.CASCADE) @JoinColumn(
        name = "user_id",
        nullable = false,
        updatable = false
    )
    @ManyToOne(fetch = FetchType.LAZY)
    val user: Users? = null,
) {
    @Id
    @UuidV7
    @Column(nullable = false, updatable = false, length = 16)
    val profileId: UUID? = null

    @CreatedDate
    @Column(nullable = false, updatable = false)
    val createdAt: LocalDateTime? = null

    @LastModifiedDate
    @Column(nullable = false)
    var updatedAt: LocalDateTime? = null

    fun update(profileInfo: A11yProfileInfo) {
        this.profileInfo = profileInfo
    }
}
